presto-common/src/main/java/com/facebook/presto/common/block/DictionaryBlock.java

```diff
@@ -401,14 +401,18 @@ public Block copyPositions(int[] positions, int offset, int length)
         Map<Integer, Integer> oldIndexToNewIndex = new HashMap<>();
         int[] newIds = new int[length];
 
+        // Using a boxed integer to avoid repeated boxing.
+        Integer nextIndex = 0;
```

Comment:
using primitive types whenever possible: `int`

Reply:
putIfAbsent takes in Object, Object. Declaring this as primitive int, will box this on the fly. But if no new keys are created the boxing can be done once and reused, so declared this as boxed integer.